spinoso-string/src/enc/binary/mod.rs

```diff
@@ -196,7 +196,15 @@ impl BinaryString {
     pub fn get_char_slice(&self, range: Range<usize>) -> Option<&'_ [u8]> {
         let Range { start, end } = range;
 
-        self.inner.get(start..end).or_else(|| self.inner.get(start..))
+        self.inner.get(start..end).or_else(|| {
+            if start > self.inner.len() {
+                None
+            } else if end <= start {
+                Some(&b""[..])
```

Comment:
This removes a few symbols to make this line read clearer.

```suggestion
                Some(&[])
```

Reply:
Fixed [here](https://github.com/artichoke/artichoke/pull/2102/files#diff-9a6cae435431760640daec73e708f6067634165194ea91e3c7f63213f8b48c28R203)